packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -0,0 +1,54 @@
+import vscode from "vscode";
+
+const COMMAND = "CodeActionProvider.Custom.OpenUrl";
+
+export function createCodeActionProvider() {
+  return vscode.languages.registerCodeActionsProvider(
+    "typespec",
+    new ExtensionCodeActionProvider(),
+    {
+      providedCodeActionKinds: ExtensionCodeActionProvider.providedCodeActionKinds,
+    },
+  );
+}
+
+export function createCommandOpenUrl() {
+  return vscode.commands.registerCommand(COMMAND, (url: string) =>
+    vscode.env.openExternal(vscode.Uri.parse(url)),
+  );
+}
+
+/**
+ * Provides code actions corresponding to diagnostic problems.
+ */
+export class ExtensionCodeActionProvider implements vscode.CodeActionProvider {
+  public static readonly providedCodeActionKinds = [vscode.CodeActionKind.QuickFix];
+
+  provideCodeActions(
+    _document: vscode.TextDocument,
+    _range: vscode.Range | vscode.Selection,
+    context: vscode.CodeActionContext,
+    _token: vscode.CancellationToken,
+  ): vscode.CodeAction[] {
+    // for each diagnostic entry that has the matching `code`, create a code action command
+    return context.diagnostics
+      .filter((diagnostic) => typeof diagnostic.code === "object" && "target" in diagnostic.code)
```

Comment:
Shouldn't we check whether the diagnostic is for typespec and has url defined here? please review the code change to double check all the sample code has been updated properly.